From 80b28b41031110bc9cbc534c91505e5ee269eac9 Mon Sep 17 00:00:00 2001
From: Chris Sides <Christopher.Sides@hpe.com>
Date: Tue, 6 Dec 2022 16:24:52 -0600
Subject: [PATCH] add entity manager config for hpe's dl360 poc

Enables initial support for the dl360 poc based on internal hpe
builds- namely sensor and fan handling.

Tested: Confirmed that HPE Fans and sensor records appear in webUI when paired with appropriate hardware

Change-Id: I34355ac829a183a169019cff331cf68bd5dd1f2d
Signed-off-by: Chris Sides <Christopher.Sides@hpe.com>
---
 configurations/hpe_dl360g10_baseboard.json | 330 +++++++++++++++++++++
 meson.build                                |   1 +
 2 files changed, 331 insertions(+)
 create mode 100644 configurations/hpe_dl360g10_baseboard.json

diff --git a/configurations/hpe_dl360g10_baseboard.json b/configurations/hpe_dl360g10_baseboard.json
new file mode 100644
index 0000000..290fa8f
--- /dev/null
+++ b/configurations/hpe_dl360g10_baseboard.json
@@ -0,0 +1,330 @@
+{
+    "Exposes": [
+        {
+            "Address": "0x4c",
+            "Bus": "5",
+            "Name": "Board Temp",
+            "Name1": "Ambient Temp",
+            "PowerState": "On",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 75
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower non critical",
+                    "Severity": 0,
+                    "Value": 5
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 0
+                }
+            ],
+            "Type": "EMC1412"
+        },
+        {
+            "Address": "0x1c",
+            "Bus": "5",
+            "Name": "PCIe Zone 1 Temp",
+            "Name1": "PCIe Zone 2 Temp",
+            "Name2": "PCIe Zone 3 Temp",
+            "Name3": "PCIe Zone 4 Temp",
+            "PowerState": "On",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 75
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower non critical",
+                    "Severity": 0,
+                    "Value": 5
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 0
+                }
+            ],
+            "Type": "EMC1414"
+        },
+        {
+            "Name": "GXP Temp",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 75
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower non critical",
+                    "Severity": 0,
+                    "Value": 5
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 0
+                }
+            ],
+            "Type": "GxpTempSensor"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 1",
+                "Pwm": 0,
+                "Tachs": []
+            },
+            "Index": 0,
+            "Name": "Fan 1",
+            "Type": "HPEFan"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 2",
+                "Pwm": 1,
+                "Tachs": []
+            },
+            "Index": 1,
+            "Name": "Fan 2",
+            "Type": "HPEFan"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 3",
+                "Pwm": 2,
+                "Tachs": []
+            },
+            "Index": 2,
+            "Name": "Fan 3",
+            "Type": "HPEFan"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 4",
+                "Pwm": 3,
+                "Tachs": []
+            },
+            "Name": "Fan 4",
+            "Type": "HPEFan"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 5",
+                "Pwm": 4,
+                "Tachs": []
+            },
+            "Index": 4,
+            "Name": "Fan 5",
+            "Type": "HPEFan"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 6",
+                "Pwm": 5,
+                "Tachs": []
+            },
+            "Index": 5,
+            "Name": "Fan 6",
+            "Type": "HPEFan"
+        },
+        {
+            "Connector": {
+                "Name": "Fan Connector 7",
+                "Pwm": 6,
+                "Tachs": []
+            },
+            "Index": 6,
+            "Name": "Fan 7",
+            "Type": "HPEFan"
+        },
+        {
+            "Class": "fan",
+            "FFGainCoefficient": 1,
+            "FFOffCoefficient": 0,
+            "ICoefficient": 0,
+            "ILimitMax": 0,
+            "ILimitMin": 0,
+            "Inputs": [
+                "Fan 1",
+                "Fan 2",
+                "Fan 3",
+                "Fan 4",
+                "Fan 5",
+                "Fan 6",
+                "Fan 7"
+            ],
+            "Name": "Fan 1-7",
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 30,
+            "Outputs": [
+                "Pwm 0",
+                "Pwm 1",
+                "Pwm 2",
+                "Pwm 3",
+                "Pwm 4",
+                "Pwm 5",
+                "Pwm 6"
+            ],
+            "PCoefficient": 0,
+            "PositiveHysteresis": 0,
+            "SlewNeg": 0,
+            "SlewPos": 0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "FFGainCoefficient": 0,
+            "FFOffCoefficient": 0,
+            "ICoefficient": -4.64,
+            "ILimitMax": 100,
+            "ILimitMin": 30,
+            "Inputs": [
+                "DTS CPU1"
+            ],
+            "Name": "DTS CPU1",
+            "NegativeHysteresis": 5,
+            "OutLimitMax": 100,
+            "OutLimitMin": 30,
+            "Outputs": [],
+            "PCoefficient": -0.15,
+            "PositiveHysteresis": 0,
+            "SetPoint": -25,
+            "SetPointOffset": "WarningHigh",
+            "SlewNeg": -1,
+            "SlewPos": 0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "FFGainCoefficient": 0,
+            "FFOffCoefficient": 0,
+            "ICoefficient": -0.2,
+            "ILimitMax": 70,
+            "ILimitMin": 10,
+            "Inputs": [
+                "Ambient Temp"
+            ],
+            "Name": "Ambient Temp",
+            "OutLimitMax": 70,
+            "OutLimitMin": 10,
+            "Outputs": [],
+            "PCoefficient": -3,
+            "SetPoint": 25,
+            "SlewNeg": -1,
+            "SlewPos": 0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "FailSafePercent": 100,
+            "MinThermalOutput": 10,
+            "Name": "Zone 1",
+            "Type": "Pid.Zone"
+        },
+        {
+            "Address": "0x30",
+            "Bus": 0,
+            "CpuID": 1,
+            "DtsCritOffset": -2,
+            "Name": "CPU 1",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Label": "DIMM",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 99
+                },
+                {
+                    "Direction": "greater than",
+                    "Label": "DIMM",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 89
+                }
+            ],
+            "Type": "XeonCPU"
+        },
+        {
+            "Address": "0x31",
+            "Bus": 0,
+            "CpuID": 2,
+            "DtsCritOffset": -2,
+            "Name": "CPU 2",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Label": "DIMM",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 99
+                },
+                {
+                    "Direction": "greater than",
+                    "Label": "DIMM",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 89
+                }
+            ],
+            "Type": "XeonCPU"
+        }
+    ],
+    "Name": "HPE DL360 G10 POC Server",
+    "Probe": [
+        "xyz.openbmc_project.FruDevice({'PRODUCT_MANUFACTURER': 'Hewlett Packard Enterprise'})",
+        "AND",
+        "xyz.openbmc_project.FruDevice({'SERVER_ID': '0x0204'})"
+    ],
+    "Type": "Chassis",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "Manufacturer": "$PRODUCT_MANUFACTURER",
+        "Model": "DL360 Gen10",
+        "PartNumber": "$PCA_PART_NUMBER",
+        "SerialNumber": "$PCA_SERIAL_NUMBER"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "HPE-OBMC-001"
+    }
+}
diff --git a/meson.build b/meson.build
index f590b79..b27ff67 100644
--- a/meson.build
+++ b/meson.build
@@ -110,6 +110,7 @@ configs = [
     'flextronics_s-1100adu00-201_psu.json',
     'greatlakes.json',
     'greatlakes_nic_mellanox.json',
+    'hpe_dl360g10_baseboard.json',
     'ibm_1000w_cffps.json',
     'ibm_1400w_cffps.json',
     'ibm_1600w_cffps.json',
